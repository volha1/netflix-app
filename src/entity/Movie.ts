type Movie = {
  readonly id: string;
  readonly title: string;
  readonly year: string;
  readonly imgPath: string;
  readonly genres: Array<string>;
};

export default Movie;
