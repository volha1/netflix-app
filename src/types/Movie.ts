type Movie = {
  readonly id: string;
  readonly title: string;
  readonly voteAverage: string;
  readonly releaseDate: string;
  readonly imgPath: string;
  readonly genres: Array<string>;
  readonly overview: string;
  readonly runtime: number;
};

export default Movie;
