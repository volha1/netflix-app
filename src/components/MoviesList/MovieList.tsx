/* eslint-disable @typescript-eslint/no-explicit-any */
import React, { Dispatch, ReactElement, SetStateAction } from 'react';
import { useSelector } from 'react-redux';
import MovieCard from '../MovieCard';
import Movie from '../../types/Movie';
import './style.scss';

type MovieSearchProps = { movie: string };
type MoviesListProps = {
  movies: Movie[];
  setParams: Dispatch<SetStateAction<MovieSearchProps>>;
};

const MoviesList = ({ movies, setParams }: MoviesListProps): ReactElement | null => {
  const loadingStatus = useSelector((state: any) => {
    return state.moviesReducer.loadingStatus;
  });

  if (!movies) {
    return null;
  }

  return (
    <div className="movies-list-wrapper content">
      <p className="counter">{loadingStatus ? 'Loading...' : `${movies.length} movies found`}</p>
      {movies.length > 0 && (
        <div className="cards-list">
          {movies.map((movie: Movie) => {
            return <MovieCard key={movie.id} movie={movie} onClick={setParams} />;
          })}
        </div>
      )}
    </div>
  );
};

export default MoviesList;
