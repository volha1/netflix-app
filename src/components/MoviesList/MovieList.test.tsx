import React from 'react';
import renderer from 'react-test-renderer';
import * as reduxHooks from 'react-redux';

import MovieList from './MovieList';
import Movie from '../../types/Movie';

jest.mock('react-redux');

describe('MovieList test', () => {
  const setParams = jest.fn();
  const movies = [] as Movie[];

  it('MovieList snapshot', () => {
    jest.spyOn(reduxHooks, 'useSelector').mockReturnValue([]);
    const component = renderer.create(<MovieList setParams={setParams} movies={movies} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
