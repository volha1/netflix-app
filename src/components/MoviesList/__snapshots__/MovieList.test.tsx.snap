// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList test MovieList snapshot 1`] = `
<div
  className="movies-list-wrapper content"
>
  <p
    className="counter"
  >
    Loading...
  </p>
</div>
`;
